code/game/machinery/computer/tram_controls.dm

```diff
@@ -15,14 +17,35 @@
 	//find the tram, late so the tram is all... set up so when this is called? i'm seriously stupid and 90% of what i do consists of barely educated guessing :)
 	find_tram()
 
+/obj/machinery/computer/tram_controls/Destroy()
+	for(var/obj/structure/industrial_lift/tram/tram_finder in GLOB.lifts)
+		tram_finder.console = null
+	return ..()
+
+/**
+ * Finds the tram to set the console to
+ *
+ * Locates tram parts in the lift global list and links the console var
+ * on the tram parts to the console itself. Essentially just another
+ * safety measure against bad tram movement.
+ */
 /obj/machinery/computer/tram_controls/proc/find_tram()
-	var/obj/structure/industrial_lift/tram/tram_struct = locate(/obj/structure/industrial_lift/tram) in GLOB.lifts
-	tram_part = tram_struct //possibly setting to something null, that's fine, but
+	var/obj/structure/industrial_lift/tram/central/tram_loc = locate() in GLOB.lifts
+	tram_part = tram_loc //possibly setting to something null, that's fine, but
 	tram_part.find_our_location()
+	for(var/obj/structure/industrial_lift/tram/tram_finder in GLOB.lifts)
+		tram_finder.console = src
```

Comment:
What's the purpose of this?

Reply:
the initial assumption with the tram getting invalid destinations was multi-inputs and since the first check on the tram structure itself didn't seem to work because it wasn't early enough, so it was placed on the console before any movement is ever called on the tram itself

this pr is a bit scuffed ngl because it was a scramble to actually fully fix the tram blender and there's like 3-4 safety checks that are now irrelevant since the actual main cause was sorted.

Reply:
I take it it's unneeded now?

Reply:
It is and I removed it in the latest commit